swift/swift.go

```diff
@@ -453,16 +455,54 @@ func (o *FsObjectSwift) Open() (in io.ReadCloser, err error) {
 	return
 }
 
+func Min(x, y int64) int64 {
+    if x < y {
+        return x
+    }
+    return y
+}
+
 // Update the object with the contents of the io.Reader, modTime and size
 //
 // The new object may have been created if an error is returned
 func (o *FsObjectSwift) Update(in io.Reader, modTime time.Time, size int64) error {
+	segmentSize, err := strconv.ParseInt(fs.ConfigFile.MustValue(o.swift.Name(), "segment-size"), 10, 64)
+	if err != nil {
+		return err
+	}
 	// Set the mtime
 	m := swift.Metadata{}
 	m.SetModTime(modTime)
-	_, err := o.swift.c.ObjectPut(o.swift.container, o.swift.root+o.remote, in, true, "", "", m.ObjectHeaders())
-	if err != nil {
-		return err
+	if size > segmentSize {
```

Comment:
I'd prefer to have a another command line variable say `--swift-upload-cutoff` which states the size that we switch to chunked uploading, rather than just using the segment size.